vispy/visuals/markers.py

```diff
@@ -30,8 +29,31 @@
 varying float v_antialias;
 
 void main (void) {
-    $v_size = a_size * u_px_scale * u_scale;
-    v_edgewidth = a_edgewidth * float(u_px_scale);
+
+    if (u_scale < 0) {
+        // left-right vector in the canvas
+        vec4 vert_on_doc = $viewtransformf(vec4(a_position, 1.0));
+        vec4 shifted_vert_on_doc = vert_on_doc + vec4(100, 0, 0, 0);
+    
+        vec4 shifted_vert = $viewtransformi(shifted_vert_on_doc);
+        shifted_vert = (shifted_vert / shifted_vert.w);
```

Comment:
Are you sure about this `/ shifted_vert.w`? As far as I can tell, it should not be here. (See wobblyness comments in #2204). You already divide by `w` when calculating the radius in px later.